### ISSUE TITLE
Editor Plugin: adding a node to a child node of the root node of the currently open scene won't show up in scene tree

### ISSUE THREAD

**Issue by mbrlabs** [godotengine/godot#37144]
**Godot version:**
3.2.1

**OS/device including version:**
Manjaro Linux

**Issue description:**
I am developing an editor plugin which is added to the dock. Essentially you click a button and nodes will be added to the currently open scene. Now when you a node (via the plugin) to the root node of the current scene everything works as expected (scene tree gets updated + the node renders in the editor). But when you add the new node as child of a child node of the root node, the new node does not show up in the scene tree (but will be rendered and can also be moved by moving the parent). If you save the scene and reload it, the new node is gone.

Adding a new node to the root node (works):
```
func _add_node(node: Node) -> void:
	var scene = get_tree().get_edited_scene_root()
	if scene == null:
		printerr("Failed to add node %s: No scene open" % node.name)
	else:
		scene.add_child(node)
		node.owner = scene
``` 

Adding new node as child of child node of root node (won't show up in scene tree):
```
func _add_node(node: Node) -> void:
	var scene = get_tree().get_edited_scene_root()
	if scene == null:
		printerr("Failed to add node: No scene open")
		return

	var container = scene.get_node("Container")
	if container == null:
		printerr("No Container node found in current scene")
		return
	
	container.add_child(node)
	node.owner = container
```

**Steps to reproduce:**
- Create an editor plugin in the dock 
- Add a new node to a child of the root node in the currently open scene via the plugin
- New node does not show up in in scene tree but it gets rendered

**Minimal reproduction project:**
Working on it

**Comment by BigRed-118** [comment]
It seems to add in the new node under container in the scene tree if you make the owner the scene tree instead of container

    func _add_node(node: Node) -> void:
	    var scene = get_tree().get_edited_scene_root()
	    if scene == null:
		    printerr("Failed to add node: No scene open")
		    return

	    var container = scene.get_node("Container")
	    if container == null:
		    printerr("No Container node found in current scene")
		    return
	
	    container.add_child(node)
	    node.set_owner(scene)

![image](https://user-images.githubusercontent.com/52541536/77192419-8c5b4e00-6aaa-11ea-962b-93af516c1ca9.png)

**Comment by BigRed-118** [comment]
@akien-mga This is similar to the discussion in #34739, but the update to the add_child documentation (#34747) doesn't clarify that the owner should be set as the scene tree node, regardless of who the parent of the child is. Is this the expected behavior? If it is I can update the documentation to clarify as such.

**Comment by mbrlabs** [comment]
You're right, that works. But you only have to set the owner manually if you add a node from outside the scene tree (i.e. a plugin).

According to the [Node.owner](https://docs.godotengine.org/en/stable/classes/class_node.html#class-node-property-owner) docs the owner of a node can be any node above it in the hirachy:

> A node can have any other node as owner (as long as it is a valid parent, grandparent, etc. ascending in the tree).

So my code should work.

**Comment by BigRed-118** [comment]
Yeah you're right. I would think that it should still show up in the scene tree even if its owner is not the root node.

**Comment by BigRed-118** [comment]
Looking into this further, when a node is added through the scene tree dock, the owner is assigned to be the root node. 
![image](https://user-images.githubusercontent.com/52541536/77219109-682a5c00-6b00-11ea-8ff9-7f10302dfc99.png)



The area that's masking the nodes with owners other than the root node is in editor\scene_tree_editor.cpp, SceneTreeEditor::_add_nodes.

    // only owned nodes are editable, since nodes can create their own (manually owned) child nodes,
    // which the editor needs not to know about.

	bool part_of_subscene = false;

	if (!display_foreign && p_node->get_owner() != get_scene_node() && p_node != get_scene_node()) {

		if ((show_enabled_subscene || can_open_instance) && p_node->get_owner() && (get_scene_node()->is_editable_instance(p_node->get_owner()))) {

			part_of_subscene = true;
			//allow
		} else {
			return false;
		}
	} else {
		part_of_subscene = p_node != get_scene_node() && get_scene_node()->get_scene_inherited_state().is_valid() && get_scene_node()->get_scene_inherited_state()->find_node_by_path(get_scene_node()->get_path_to(p_node)) >= 0;
	}

The leading comment seems like the intent is for the manually owned child nodes to not be seen by the editor? If display_foreign was set to true, everything would be shown. Otherwise, logic would need to be fit in here for the case where you add a node from outside the scene tree.

**Comment by corjohnson** [comment]
`Using Godot 3.2.2`

I had a problem similar to this while working on importing a cell fractured cube from blender. The scene imported had 50 or so objects, but they were all just mesh instances. I tried to write an editor script that did two steps: add an area for each of those cell fractures, named `cell_01_area` or similar; and then it would create a tri-mesh sibling from the `cell_01` mesh instance & create a collision shape for collisions. 

I'd run the script and the areas would be created, and the collision shapes would be created as well, but I could never see them. Perhaps this isn't the same exact kind of issue, but it seems related?

I'm attaching the relevant files from my project if there's any interest, but I'm not sure I'm familiar enough with the code to debug this

I had imported the fbx file as instanced scenes or whatever the non-readonly version is

[cell_collision_gen.zip](https://github.com/godotengine/godot/files/4663007/cell_collision_gen.zip)

**Comment by naturally-intelligent** [comment]
Having the same issue with EditorScript (3.4.5). Adding any kind of nesting works temporarily in the editor view, but just doesn't show up in the scene tree. Also the nested nodes disappear if launched or saved.

`# works, shows in the editor and scene tree
var parent = Node2D.new()
get_scene().add_child(parent)
parent.set_owner(get_scene())
# shows in editor, but doesn't show as child in scene tree
var sprite = Sprite.new()
sprite.texture = ...
parent.add_child(sprite)
sprite.set_owner(parent)`

**Comment by bojidar-bg** [comment]
Does it work if you change the last line to <code>sprite.set_owner(<i>get_scene()</i>)</code>?

**Comment by naturally-intelligent** [comment]
> Does it work if you change the last line to `sprite.set_owner(get_scene())`?

Yes. 

I thought I had tried that, but I double checked, on your question, and it does work!

**Comment by Neo-Ciber94** [comment]
What's the `get_scene` equivalent on Godot 4.5?

On my editor plugin I tried, but got `ERROR: Invalid owner. Owner must be an ancestor in the tree.`

```gdscript
class_name MainPanel extends Control

func some_func():
   # ...
   icons_container.add_child(icon_image)
   icon_image.owner = get_tree().edited_scene_root
```
